patroni/postgresql.py

```diff
@@ -698,31 +690,28 @@ def start(self, block_callbacks=False):
 
         self.set_state('starting')
 
-        env = {'PATH': os.environ.get('PATH')}
-        # pg_ctl will write a FATAL if the username is incorrect. exporting PGUSER if necessary
-        if 'username' in self._superuser and self._superuser['username'] != os.environ.get('USER'):
-            env['PGUSER'] = self._superuser['username']
-
         self._write_postgresql_conf()
         self.resolve_connection_addresses()
 
-        options = ' '.join("--{0}='{1}'".format(p, self._server_parameters[p]) for p, v in self.CMDLINE_OPTIONS.items()
+        options = ' '.join('--{0}="{1}"'.format(p, self._server_parameters[p]) for p, v in self.CMDLINE_OPTIONS.items()
```

Comment:
I think server parameter values need to be escaped for shell here. Are there any sanity checks for server parameter keys in Patroni? If not, that might also have some security implications. Having write permissions for Patroni configuration will probably allow for arbitrary code execution no matter what we do, but it shouldn't be this easy :)